### DISCUSSION TITLE
Better language support for C++ would be appreciated!

### DISCUSSION THREAD

**Discussion by cbedhoam620** [godotengine/godot-proposals#8290] [Scripting]
Now use GDExtension, and then manually use godot-cpp to create project templates, a little too cumbersome, I hope to be able to build-in language support, of course, we do not expect to be able to directly in the Godot built-in editor can write cpp code, but we want to be able to Godot through the Node or control right-click on the attached script, can be popped up in the " Set script for node" dialog box, the language drop-down list box, you can add the option of C++, and automatically configure the godot-cpp project template to simplify the development.
![image](https://github.com/godotengine/godot-proposals/assets/117166289/21fd28c7-612b-4b1b-ac3d-fbca942cf936)

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/573.
